### ISSUE TITLE
Spotify Link Issue Still Persisting with DNS Protection

### ISSUE THREAD

**Issue by blades808** [AdguardTeam/AdGuardSDNSFilter#1489]
### Prerequisites

- [X] This site DOES NOT contains sexually explicit material;
- [X] The problem occurs only when using AdGuard DNS or DNS filtering with AdGuard DNS filter, it is not caused by other ad blockers;
- [X] You're using an up-to-date version of AdGuard DNS filter;
- [X] Browser version is up-to-date;
- [X] If a website or an app is broken, disabling AdGuard DNS filter resolves the issue.

### What DNS server do you use?

AdGuard public DNS

### Version

AdGuard for iOS 4.5.0

### What DNS upstream(s) do you use in AdGuard apps or AdGuard Home?

Adguard-dns.io 

### What DNS filters do you have enabled?

No DNS filters are enabled

### What browser or app do you use?

Other app

### Which device type do you use?

Mobile

### What type of problem have you encountered?

Website or app doesn't work properly

### Where did you encounter the problem?

https://spotify.link/OvgpA6AetDb

### Add your comment and screenshots

This issue specifically and only relates to the DNS protection part of the iPhone app. Turning on and off Safari and advanced protection does not affect this issue in any way, and even when those services are on, when I disable the DNS protection part of the app the link works as intended.

I noticed this issue when sending and receiving songs via Spotify links in Instagram and iMessage. When DNS protection is enabled, even with no extra DNS filters, all that happens is the Spotify app opens to the most recent page. The intended behavior is that the link will directly to a specific song. Somehow it looks that when the link is being run through the DNS protection, the specific pointer is getting lost.

If I disable DNS protection and restart my applications, then it will quickly resume the intended behavior, and when I enable DNS protection, I will no longer be able to link to a specific song. This means that the intended behavior of quickly and easily sharing content via these applications is effectively broken when using DNS protection in AdGuard.

### Privacy

- [X] I agree to follow this condition

**Comment by Alex-302** [comment]
Hi. Cannot reproduce. Please send report from Feedback menu.

**Comment by blades808** [comment]
> Hi. Cannot reproduce. Please send report from Feedback menu.

Gosh you know I tried the issue first like you described through the app and they told me to come here if it was still a problem. Which it is! I explored further and the issue is still persisting on my iPhone. I wonder why it’s happening.

The issue even resolves when I switch from “Adguard DNS” to “Adguard DNS Non Filtering” and comes back when I switch back to “Adguard DNS”

obviously the version that filters ads in outside apps is preferable. 

I will keep exploring to see why this might be breaking .

thank you!
